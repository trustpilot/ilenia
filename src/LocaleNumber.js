import withTranslations from './withTranslations';
import PropTypes from 'prop-types';

const LocaleNumber = ({ number, locale='en-US', maxDecimals }) => {
  try {
    const options = typeof maxDecimals === 'number' ?
      { maximumFractionDigits: maxDecimals } : {};
    return number.toLocaleString(locale, options);
  } catch (error) {
    return number;
  }
};

LocaleNumber.propTypes = {
  number: PropTypes.number.isRequired,
  maxDecimals: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
};

export default withTranslations(LocaleNumber);
