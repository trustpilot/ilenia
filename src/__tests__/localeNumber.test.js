import React from 'react';
import renderer from 'react-test-renderer';

import LocalizationProvider from '../LocalizationProvider';
import LocaleNumber from '../LocaleNumber';

test('Renders a localized number with a correct locale', () => {
  const component = renderer.create(
    <LocalizationProvider locale="en-US" translations={{}}>
      <LocaleNumber number={1000000} />
    </LocalizationProvider>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a localized number with an incorrect locale', () => {
  const component = renderer.create(
    <LocalizationProvider locale="abcdefghijklmnopqrstuvwxyz" translations={{}}>
      <LocaleNumber number={1000000} />
    </LocalizationProvider>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders truncated and localized number when max decimal is set', () => {
  const component = renderer.create(
    <LocalizationProvider locale="en-US" translations={{}}>
      <LocaleNumber number={9.392} maxDecimals={2} />
    </LocalizationProvider>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders non decimal number when max decimal is set to zero', () => {
  const component = renderer.create(
    <LocalizationProvider locale="en-US" translations={{}}>
      <LocaleNumber number={13.2452} maxDecimals={0} />
    </LocalizationProvider>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders only none zero decimals when maxDecimals is set', () => {
  const component = renderer.create(
    <LocalizationProvider locale="en-US" translations={{}}>
      <LocaleNumber number={3.2} maxDecimals={2} />
    </LocalizationProvider>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
