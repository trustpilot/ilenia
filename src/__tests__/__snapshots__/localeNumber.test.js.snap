// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a localized number with a correct locale 1`] = `"1,000,000"`;

exports[`Renders a localized number with an incorrect locale 1`] = `"1000000"`;

exports[`Renders non decimal number when max decimal is set to zero 1`] = `"13"`;

exports[`Renders only none zero decimals when maxDecimals is set 1`] = `"3.2"`;

exports[`Renders truncated and localized number when max decimal is set 1`] = `"9.39"`;
